boxoffice/views/order.py

```diff
@@ -433,6 +433,55 @@ def cancel_line_item(line_item):
     return make_response(jsonify(status='ok', result={'message': 'Ticket cancelled', 'cancelled_at': date_format(line_item.cancelled_at)}), 200)
 
 
+def process_partial_refund_for_order(order, form_dict):
+    form = OrderRefundForm.from_json(form_dict, meta={'csrf': False})
+    if form.validate():
+        requested_refund_amount = form.amount.data
+        order_payment_transactions = order.get_payment_transactions().all()
+        if not order_payment_transactions:
+            return make_response(jsonify(status='error', error='free_order',
+                error_description='Refunds can only be issued for paid orders'), 403)
+
+        total_paid_amount = sum([order_transaction.amount
+            for order_transaction in order_payment_transactions])
+
+        total_refunded_amount = sum([order_transaction.amount
+            for order_transaction in order.get_refund_transactions()])
+
+        if (total_refunded_amount + requested_refund_amount) >= total_paid_amount:
+            return make_response(jsonify(status='error', error='excess_partial_refund',
+                error_description='Invalid refund amount, must be lesser than net amount paid for the order'), 403)
+
+        payment = OnlinePayment.query.filter_by(order=order,
+            pg_payment_status=RAZORPAY_PAYMENT_STATUS.CAPTURED).one()
+        rp_resp = razorpay.refund_payment(payment.pg_paymentid, requested_refund_amount)
+        if rp_resp.status_code == 200:
+            transaction = PaymentTransaction(order=order, transaction_type=TRANSACTION_TYPE.REFUND,
+                online_payment=payment, amount=requested_refund_amount, currency=CURRENCY.INR,
+                internal_note=form.internal_note.data, note_to_user=form.note_to_user.data)
+            db.session.add(transaction)
+            db.session.commit()
+            send_order_refund_mail.delay(order.id, transaction.amount, transaction.note_to_user)
+            return make_response(jsonify(status='ok', result={'message': 'Refund processed for order'}), 200)
```

Comment:
Better to send the latest order amount to the client so that the UI could be updated.